src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/zip/ZipStepExecution.java

```diff
@@ -75,10 +75,13 @@ static class ZipItFileCallable extends AbstractFileCallable<Integer> {
         final String exclude;
         final boolean overwrite;
 
-        public ZipItFileCallable(String glob, String exclude, boolean overwrite) {
+        private boolean defaultExcludes = true;
```

Comment:
Making this field `final` would make it both consistent with the other fields in this class and easier to reason about (mutable state is generally harder to reason about than immutable state).

Reply:
@basil If i make it final, how can i assign the default value as true as by default we want not to includes patterns which are excludes by ant java pattern.

Reply:
Done. can you review?